aten/src/ATen/autocast_mode.cpp

```diff
@@ -257,65 +287,65 @@ TORCH_LIBRARY_IMPL(_, Autocast, m) {
 }
 
 TORCH_LIBRARY_IMPL(aten, Autocast, m) {
```

Comment:
@ezyang Should this say AutocastCUDA? Simple question, but the answer seems a bit of a rabbit hole.  Here's what I _think_ is happening (some of the tech is new since I last looked at dispatch):
[getAutocastRelatedKeySetFromBackend](https://github.com/pytorch/pytorch/pull/57386/files#diff-da6c1aa633224c76f5ae6b4e80676dadbb24bac021770ee01d5560d2db8ffb92R76)'s only purpose, afaict is to set the AutocastCUDA or AutocastCPU, as appropriate, on every single CUDA or CPU tensor.
`set_autocast_enabled`, then, doesn't _include_ `AutocastCUDA`, rather, it [suppresses the exclusion of AutocastCUDA](https://github.com/pytorch/pytorch/pull/57386/files#diff-b99d3aff6a9039f8abada80fa8c2647a85d8248fa22b64310a0cc9ca620dad62R20). Overall (IIUC) the approach "always enable the right Autocast key on tensors of a particular device, but also exclude the key by default and include it when needed by suppressing the exclusion" seems good: for example, it prevents an arglist containing only CPU tensors from ever dispatching into the AutocastCUDA layer (even though the layer wouldn't touch the tensors, the useless dispatch trip is good to avoid).

Anyway, AutocastCUDA aliases to Autocast, so effectively "Autocast" in TORCH_LIBRARY_IMPL above is "AutocastCUDA" already. If my reading of the flow is right, I don't see a reason _not_ to make it explicitly "AutocastCUDA", then delete the Autocast = AutocastCUDA line. But maybe there's some other reason for keeping Autocast and AutocastCUDA separate that I don't understand.

Reply:
Yes it should. We need some cleanup to s/Autocast/AutocastCUDA/ and then expunge all Autocast from downstream libs and we're good.

Reply:
I think I'll make that change in a separate PR (not because it would be hard in this PR, but to separate any consequences, the CPU-specific diffs here don't deserve reversion risk from a CUDA-specific change).

Reply:
Yup agreed